source/common/network/address_impl.cc

```diff
@@ -377,7 +380,14 @@ Api::SysCallIntResult PipeInstance::bind(int fd) const {
     unlink(address_.sun_path);
   }
   auto& os_syscalls = Api::OsSysCallsSingleton::get();
-  return os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  auto bind_result = os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  if (mode != 0 and !abstract_namespace_ and bind_result.rc_ == 0) {
+    auto set_permissions = os_syscalls.chmod(address_.sun_path, mode);
```

Comment:
We have `fd` already, so why not using `fchmod`?

Reply:
AFAICT they would be equivalent

Reply:
I believe fchmod is slightly more performant (without resolving path), and resilient to race condition when the pipe is moved after opening.

Reply:
I agree `fchmod` is more performant but it returns `EINVAL` on BSD based systems for socket fds. `chmod` seems like the most portable option here

Reply:
ok that's fair.